go/feast/basefeatureview.go

```diff
@@ -0,0 +1,43 @@
+package feast
+
+import (
+	"errors"
+	"fmt"
+	"github.com/feast-dev/feast/go/protos/feast/core"
+)
+
+type BaseFeatureView struct {
```

Comment:
I think I may have missed the addition of BaseFeatureView to the Python codebase, but is there any reason we can't have purely a FeatureView and NewFeatureView function?

Reply:
I think Python codebase has BaseFeatureView and I'm based off of that. When ODFVs and Request FVs are supported, all of them can have projections and share function withProjection() so I thought this would be useful to share certain properties of these 3 types of FV

Reply:
Yea the question wasn't so much directed at you. I understand we want a parent. I guess the problem is that we don't have a nice name for the parent of FV and ODFV.